### PR TITLE
PostgreSQL: PGX fix multiple results handling

### PR DESCRIPTION
**Pull request by zoltanbedi**
- Added tests for handling multiple result sets, including compatible and incompatible structures, ensuring no panics occur.
- Improved `convertResultsToFrame` function to validate column compatibility and handle null values correctly.
- Introduced a new helper function `convertPostgresValue` for converting raw PostgreSQL values to appropriate Go types.
- Added comprehensive unit tests for `convertResultsToFrame` covering various scenarios including row limits and mixed result types.

**Which issue(s) does this PR fix?**:

Fixes #110450

### CODE REVIEW

File: pkg/tsdb/grafana-postgresql-datasource/postgres_pgx_test.go [hunk 1]
```diff
@@ -1398,5 +1398,124 @@ func TestIntegrationPostgresPGX(t *testing.T) {
...
+				// The frame should be properly constructed from both SELECT results
+				frame := frames[0]
+				require.Equal(t, 2, len(frame.Fields)) // id, name from both queries
+				require.Equal(t, "id", frame.Fields[0].Name)
+				require.Equal(t, "name", frame.Fields[1].Name)
+				require.Equal(t, 4, frame.Rows()) // 2 rows from first result + 2 rows from second result
+			})
+
+			t.Run("Should return error for incompatible multiple result sets", func(t *testing.T) {
+				// This query returns multiple result sets with different structures - the kind that used to cause panic
```
Comment (dgiagio) [inline pkg/tsdb/grafana-postgresql-datasource/postgres_pgx_test.go:1465]: Can we add another test to handle incompatible "number of columns" as well? It should fix the error `runtime error: index out of range [1] with length 1` and return the message `convert frame from rows error: incompatible result structure: expected X columns, got Y columns`

An example of query to trigger this case is
```sql
SELECT id, name FROM test_multi_results WHERE id = 1;
SELECT id FROM test_multi_results WHERE id = 1;
```

**Review by dgiagio — CHANGES_REQUESTED** [review]
Looks good! I think we just need one more test (see my comment) to make sure we test every case we're expecting to cover.

File: pkg/tsdb/grafana-postgresql-datasource/postgres_pgx_test.go (hunk 1 above)
Comment (zoltanbedi) [reply pkg/tsdb/grafana-postgresql-datasource/postgres_pgx_test.go:1465]: Added that test case.

**Review by dgiagio — APPROVED** [review]
